kaggle/api/kaggle_api_extended.py

```diff
@@ -86,98 +86,248 @@ class KaggleApi(KaggleApi):
                 config_data = json.load(f)
-            if value is not None:
-                config_data[name] = value
-                with open(self.config, 'w') as f:
-                    json.dump(config_data, f, indent=2)
-            if name in config_data:
-                self.config_values[name] = config_data[name]
-        except:
-            pass
-        if not quiet:
-            self.print_config_value(name, separator=' is now set to: ')
+        except FileNotFoundError:
+            config_data = {}
+
+        return config_data
+
+
+    def _write_config(self, config_data, indent=1):
+        '''write config data to file.
+        '''
+        with open(self.config, 'w') as f:
+            json.dump(config_data, f, indent=indent)
+
+
+    def set_config_value(self, name, value, quiet=False):
+        '''a client helper function to set a configuration value, meaning
+           reading in the configuration file (if it exists), saving a new
+           config value, and then writing back
+        '''
+
+        config_data = self._read_config()
+   
+        # If defined by client, set and save!
+        self._write_config(config_data)
+
+        if value is not None:
+
+            # Update the config file with the value
+            config_data[name] = value
+
+            # Update the instance with the value
+            self.config_values[name] = value
+
+            if not quiet:
+                self.print_config_value(name, separator=' is now set to: ')
+
 
     def unset_config_value(self, name, quiet=False):
-        try:
-            with open(self.config, 'r') as f:
-                config_data = json.load(f)
-            config_data[name] = None
-            with open(self.config, 'w') as f:
-                json.dump(config_data, f, indent=2)
-            self.config_values[name] = None
-        except:
-            pass
-        if not quiet:
-            self.print_config_value(name, separator=' is now set to: ')
+        '''unset a configuration value'''
 
-    def copy_config_value(self, name, values):
-        if name in values:
-            self.config_values[name] = values[name]
+        config_data = self._read_config()
+
+        # Remove it, if exists, both from loaded file and client
+
+        del self.config_values[name] # is there reason this was None?
+
+        if name in config_data:
+            del config_data[name]
+
+            self._write_config(config_data)
+
+            if not quiet:
+                self.print_config_value(name, separator=' is now set to: ')
 
     def get_config_value(self, name):
+        ''' return a config value (with key name) if it's in the config_values,
+            otherwise return None
+ 
+            Parameters
+            ==========
+            name: the config value key to get
+
+        '''
         if name in self.config_values:
             return self.config_values[name]
-        return None
 
-    def get_default_download_dir(self, *subdirs):
```

Comment:
This path was intentionally renamed to `get_default_download_dir` because while I understand the consistency with `get_config_path` and the other methods, it's also confusing between does it get the value of the configuration key path, or does it get the path of the configuration file.  

Getting the configuration directory and getting the download path need to be separate.